csl-variables.rnc

```diff
@@ -80,6 +83,8 @@ div {
     | "original-title"
     | "page"
     | "page-first"
+    | "part-number"
```

Comment:
Is there a special reason why part number is not a number variable?

Reply:
no; are we 100% sure it should be moved? I'm forgetting.

Reply:
I'm not... @adam3smith @bwiernik What do you think?

Reply:
I think it is probably here because it is similar to `page`, which also isn't a number variable. The main function of a variable being a number is so that it can be formatted as an ordinal. While usually they would be formatted as "Part 2", etc., I can imagine cases where a style might call for an ordinal form (especially in French --e.g., `2ème partie`). So making `part-number` a number makes sense to me.

Reply:
The [CSL-M schema](https://citeproc-js.readthedocs.io/en/latest/csl-m/#page-and-page-first-modification) changes `page` and `page-first` to numeric variables. I think this should be done in CSL as well. Current citeproc-js behavior on hyphen-substitution in `cs:text` elements produces undesirable output: https://discourse.citationstyles.org/t/en-dash-subsittution-in-text/1497

Relatedly, I think `locator` should also be a number?